File: DemoServer/Indexes/ManyProduct.cs
using System.Linq;
using Raven.Client.Documents.Indexes;

namespace DemoServer.Indexes
{
    public class ManyProduct : AbstractIndexCreationTask<ProductName>
    {
        public ManyProduct()
        {
            Map = products => from product in products
                              select new
                              {
                                  product
                              };
        }
    }
}

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/indexes/ManyProduct.java
package net.ravendb.demo.indexes;

import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;

public class ManyProduct extends AbstractIndexCreationTask {

    public ManyProduct() {
        map = "from product in docs.ProductNames " +
                "select new { " +
                "  product = product " +
                "}";
    }
}
